src/test/unit/algorithms/ConnectionsTest.cpp

```diff
@@ -34,31 +35,16 @@ using namespace nupic::algorithms::connections;
 
 #define EPSILON 0.0000001
 
-namespace nupic {
+namespace {
 
-  void ConnectionsTest::RunTests()
-  {
-    testCreateSegment();
-    testCreateSegmentReuse();
-    testCreateSynapse();
-    testDestroySegment();
-    testDestroySynapse();
-    testUpdateSynapsePermanence();
-    testMostActiveSegmentForCells();
-    testMostActiveSegmentForCellsNone();
-    testLeastRecentlyUsedSegment();
-    testComputeActivity();
-    testActiveSegments();
-    testNumSegments();
-    testNumSynapses();
-    testWriteRead();
-    testSaveLoad();
-  }
+  // Forward definitions
+  void setupSampleConnections(Connections &connections);
```

Comment:
Probably simpler to just implement the functions up here rather than forward declaring them and defining them at the end.